src/org/traccar/events/AlertEventHandler.java

```diff
@@ -26,11 +26,16 @@
 
     @Override
     protected Map<Event, Position> analyzePosition(Position position) {
-        Object alarm = position.getAttributes().get(Position.KEY_ALARM);
-        if (alarm != null) {
-            Event event = new Event(Event.TYPE_ALARM, position.getDeviceId(), position.getId());
-            event.set(Position.KEY_ALARM, (String) alarm);
-            return Collections.singletonMap(event, position);
+        Object alarms = position.getAttributes().get(Position.KEY_ALARM);
+        Map<Event, Position> eventMap = new LinkedHashMap<>();
```

Comment:
It seems like this can be moved into the `if` block.

Reply:
Ok